src/lit-html.ts

```diff
@@ -661,6 +661,10 @@ export const reparentNodes =
      end: Node | null = null,
      before: Node | null = null): void => {
       let node = start;
+      if (before && before === end) {
+        // Prevent a really fun infinite loop.
```

Comment:
I think there are other similar cases that would cause an infinite loop still, like `before` being a preceding sibling of `end`, but a following sibling of `start`. Since we should document not to do that anyway, and this only protected against one case, I wonder if we should move the check into repeat.ts: https://github.com/PolymerLabs/lit-html/blob/master/src/lib/repeat.ts#L77

Reply:
That's where I had it originally, I'll move it back.